pkg/jx/cmd/create_jenkins_token.go

```diff
@@ -208,7 +197,8 @@ func (o *CreateJenkinsUserOptions) Run() error {
 	return nil
 }
 
-func (o *CreateJenkinsUserOptions) tryFindAPITokenFromBrowser(tokenUrl string, newTokenUrl string, userAuth *auth.UserAuth) error {
+func (o *CreateJenkinsUserOptions) getAPITokenFromREST(serverURL string, userAuth *auth.UserAuth) error {
+	newTokenURL := jenkins.JenkinsNewTokenURL(serverURL)
 	var ctx context.Context
 	var cancel context.CancelFunc
 	if o.Timeout != "" {
```

Comment:
You can create the context with a timeout.

```Go
ctx, cancel := context.WithTimeout(context.Background(), timeout)
defer cancel()
```
This will ensure that the context will be canceled automatically after the timeout expires.

Reply:
just added that line